From 3fbe93371258704e56dc741218b4faa399395f8a Mon Sep 17 00:00:00 2001
From: Paul Holzinger <pholzing@redhat.com>
Date: Mon, 4 Jul 2022 17:24:24 +0200
Subject: [PATCH 1/3] envVarValueResourceFieldRef: use int64 for value

int can be 32 or 64 bit depending on the architecture.
The total memory is int64 so we have to use int64 for the value as
well otherwise we get an overflow on 32 bit systems.

Fixes #14819

Signed-off-by: Paul Holzinger <pholzing@redhat.com>
---
 pkg/specgen/generate/kube/kube.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/pkg/specgen/generate/kube/kube.go
+++ b/pkg/specgen/generate/kube/kube.go
@@ -769,8 +769,8 @@
 	}
 
 	// k8s rounds up the result to the nearest integer
-	intValue := int(math.Ceil(value.AsApproximateFloat64() / divisor.AsApproximateFloat64()))
-	stringValue := strconv.Itoa(intValue)
+	intValue := int64(math.Ceil(value.AsApproximateFloat64() / divisor.AsApproximateFloat64()))
+	stringValue := strconv.FormatInt(intValue, 10)
 
 	return &stringValue, nil
 }
